.github/workflows/python_ci.yml

```
@@ -3,13 +3,16 @@ name: Run base code on different python versions
 jobs:
   python_run_scripts:
     strategy:
+      fail-fast: false
       matrix:
-        version: [3.9, 3.10, 3.11]
+        version: ['3.9', '3.10', '3.11']
     runs-on: ubuntu-latest
     steps:
+      - name: install mpi
+        run: sudo apt update && sudo apt-get install openmpi-bin openmpi-doc libopenmpi-dev 
       - uses: actions/checkout@v3
       - name: setup python
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.version }} # install the python version needed
-      - run: python install --editable ".[dev]"
+      - run: pip install --editable ".[dev]"
```

This probably doesn't need to be `--editable` – as you're not going to edit it here, you can install the package in its normal mode. I'd also wonder whether the `[dev]` option is necessary – perhaps the check here could be "does the package work in the way a typical user would use it, that is without the dev dependencies?"
```suggestion
      - run: pip install .
```

---

Good point. I'll test it without the `[dev]` option.